### PR TITLE
Upgrades to streamlit app

### PR DESCRIPTION
**Pull request by sultaniman**
This PR upgrades our streamlit app #459 which is launched when `dlt pipeline PIPELINE_NAME show` command is called.
It also includes better prompt to detect and install streamlit and additional packages if it is missing.

image

image

### CODE REVIEW

File: dlt/cli/helpers/dependencies.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            install_missing_package("streamlit")
+        else:
+            raise MissingDependencyException(
+                "DLT Streamlit Helpers",
+                ["streamlit"],
+                "DLT Helpers for Streamlit should be run within a streamlit app.",
+            )
+
+
+maybe_install_streamlit()
```
Comment (sultaniman) [inline dlt/cli/helpers/dependencies.py:71]: I will remove this and will call it in `streamlit_app/init.py`

**Review by rudolfix — CHANGES_REQUESTED** [review]
Let's review UI together! First take:
- the sidebar should contain the most important information like pipeline name, destination etc. maybe last load if trace is present
- I do not want to distribute images with the library
- OR we should distribute the streamlit app as dlt extra: we extract this app as seprate dependency

otherwise I'm so happy we are pushing that forward

File: dlt/cli/helpers/dependencies.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        try:
+            check_call(["pip", "install", package])
+        except CalledProcessError as e:
+            print(f"Could not install {package}")
+            sys.exit(e.returncode)
+    finally:
+        globals()[package] = importlib.import_module(package)
+
+
+def query_yes_no(question, default="yes"):
```
Comment (rudolfix) [inline dlt/cli/helpers/dependencies.py:20]: what's wrong with current `echo.py` and `clikck` library we use. this should IMO work

**Review by rudolfix — CHANGES_REQUESTED** [review]

**Comment by sultaniman** [comment]
Closing this in favor of https://github.com/dlt-hub/dlt/pull/1060 because I decided to split automatic dependency installation utilities in the separate PR.

**Pull request closed without merging**